gcore/src/exec.rs

```diff
@@ -188,6 +193,19 @@ pub fn wait() -> ! {
     unsafe { sys::gr_wait() }
 }
 
+/// Same as [`wait`], but delays handling for given specific amount of blocks.
+///
+/// NOTE: It panics, if given duration couldn't be totally payed.
+pub fn wait_for(duration: u32) -> ! {
+    unsafe { sys::gr_wait_for(duration.to_le_bytes().as_ptr()) }
+}
+
+/// Same as [`wait`], but delays handling for maximal amount of blocks
+/// that could be payed, that doesn't exceed given duration.
```

Comment:
I would ask you to provide some examples to highlight the difference between `wait_for` and `wait_no_more` calls.

Reply:
Frankly speaking, it's hard to do some interesting and useful for ecosystem demo before #1380 (and probably #1331) solved.
But there is already the core test in pallet gear, that shows how it works.